### PR TITLE
[fmt] Fix build error when building arm64-uwp

### LINKED ISSUE #16764: [fmt:arm64-uwp] Build failure
**Host Environment**
- OS: Windows
- Compiler: ninja 1.10.2

**To Reproduce**
`./vcpkg install fmt:arm64-uwp`

**Failure logs** 
-Cut and past the appropriate build messages from the console output.
```
❯ .\vcpkg.exe install spdlog:arm64-uwp
Computing installation plan...
The following packages will be built and installed:
  * fmt[core]:arm64-uwp -> 7.1.3#1
    spdlog[core]:arm64-uwp -> 1.8.0#2
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet arm64-uwp...
Could not locate cached archive: C:\Users\hanseul\AppData\Local\vcpkg\archives\b6\b63bdc464a2d77b2648dccdc4158d5740823d6d5.zip
Could not locate cached archive: C:\Users\hanseul\AppData\Local\vcpkg\archives\be\be2822ff7dbc32ea55bf395501b2b2c5accd6f84.zip
Starting package 1/2: fmt:arm64-uwp
Building package fmt[core]:arm64-uwp...
-- Using community triplet arm64-uwp. This triplet configuration is not guaranteed to succeed.
-- [COMMUNITY] Loading triplet configuration from: C:\Users\hanseul\github\telespace-framework\deps\vcpkg\triplets\community\arm64-uwp.cmake
-- Using cached C:/Users/hanseul/github/telespace-framework/deps/vcpkg/downloads/fmtlib-fmt-7bdf0628b1276379886c7f6dda2cef2b3b374f0b.tar.gz
-- Cleaning sources at C:/Users/hanseul/github/telespace-framework/deps/vcpkg/buildtrees/fmt/src/2b3b374f0b-8ea71ee8ea.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/Users/hanseul/github/telespace-framework/deps/vcpkg/downloads/fmtlib-fmt-7bdf0628b1276379886c7f6dda2cef2b3b374f0b.tar.gz
-- Applying patch fix-warning4189.patch
-- Using source at C:/Users/hanseul/github/telespace-framework/deps/vcpkg/buildtrees/fmt/src/2b3b374f0b-8ea71ee8ea.clean
-- Found external ninja('1.10.2').
-- Configuring arm64-uwp
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:105 (message):
    Command failed: ninja -v
    Working Directory: C:/Users/hanseul/github/telespace-framework/deps/vcpkg/buildtrees/fmt/arm64-uwp-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      C:\Users\hanseul\github\telespace-framework\deps\vcpkg\buildtrees\fmt\config-arm64-uwp-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:351 (vcpkg_execute_required_process)
  ports/fmt/portfile.cmake:9 (vcpkg_cmake_configure)
  scripts/ports.cmake:142 (include)
```
-Please attach any additional failure logs mentioned in the console output.
[config-arm64-uwp-out.log](https://github.com/microsoft/vcpkg/files/6161795/config-arm64-uwp-out.log)

### PR DESCRIPTION
**Pull request by JackBoosY**
Fix configure error:
```
CMake Error at CMakeLists.txt:160 (file):
  file attempted to write a file:
  C:/Users/hanseul/github/telespace-framework/deps/vcpkg/buildtrees/fmt/src/2b3b374f0b-8ea71ee8ea.clean/run-msbuild.bat
  into a source directory.
```

Fixes #16764.

### CODE REVIEW

**Review by NancyLi1013 — APPROVED** [review]

**Review by ras0219-msft — APPROVED** [review]
This LGTM, however we need to merge #16774 first and re-run CI to ensure this fixes the issue.

**Comment by ras0219-msft** [comment]
LGTM once merged from master.

**Comment by JackBoosY** [comment]
Already merged from master.

File: ports/fmt/portfile.cmake [hunk 1]
```diff
@@ -4,7 +4,9 @@ vcpkg_from_github(
     REF 7bdf0628b1276379886c7f6dda2cef2b3b374f0b # v7.1.3
     SHA512 52ea8f9d2c0cb52ec3a740e38fcdfd6a0318566e3b599bd2e8d557168642d005c0a59bc213cff2641a88fed3bb771d15f46c39035ccd64809569af982aba47aa
     HEAD_REF master
-    PATCHES fix-warning4189.patch
+    PATCHES
+        fix-warning4189.patch
+        fix-write-batch.patch
 )
 
 vcpkg_cmake_configure(
```
Comment (ras0219-msft) [inline ports/fmt/portfile.cmake:12]: Please remove the line below:
```
DISABLE_PARALLEL_CONFIGURE # with MSBuild (on UWP), fmt writes into the source directory
```

**Review by ras0219-msft — CHANGES_REQUESTED** [review]

**Review by vicroms — APPROVED** [review]

**Pull request merged**